src/main/java/fish/payara/micro/project/PayaraMicroModuleBuilder.java

```diff
@@ -0,0 +1,156 @@
+import com.intellij.openapi.application.ModalityState;
+import com.intellij.openapi.module.ModuleType;
+import com.intellij.openapi.options.ConfigurationException;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.projectRoots.Sdk;
+import com.intellij.openapi.roots.ModifiableRootModel;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.util.lang.JavaVersion;
+import fish.payara.PayaraBundle;
+import fish.payara.PayaraConstants;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.idea.maven.model.MavenArchetype;
+import org.jetbrains.idea.maven.model.MavenId;
+import org.jetbrains.idea.maven.wizards.MavenModuleBuilderHelper;
+
+import javax.swing.*;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.intellij.openapi.module.StdModuleTypes;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.openapi.vfs.LocalFileSystem;
+import static fish.payara.micro.project.PayaraMicroConstants.*;
+
+public class PayaraMicroModuleBuilder extends JavaModuleBuilder {
+
+    private final ModuleDescriptor moduleDescriptor = new ModuleDescriptor();
+    
+    @Override
+    public final String getParentGroup() {
+        return MODULE_ID;
+    }
+
+    @Override
+    public final int getWeight() {
+        return 10;
+    }
+
+    @Override
+    public ModuleType<?> getModuleType() {
+        return StdModuleTypes.JAVA;
+    }
+
+    @Override
+    public Icon getNodeIcon() {
+        return PayaraConstants.PAYARA_ICON;
+    }
+
+    @Nls(capitalization = Nls.Capitalization.Sentence)
+    @Override
+    public String getDescription() {
+        return MODULE_DESCRIPTION;
+    }
+
+    @Nls(capitalization = Nls.Capitalization.Title)
+    @Override
+    public String getPresentableName() {
+        return MODULE_TITLE;
+    }
+
+    @Nullable
+    @Override
+    public ModuleWizardStep getCustomOptionsStep(WizardContext context, Disposable parentDisposable) {
+        return new PayaraMicroProjectWizardStep(moduleDescriptor, context);
+    }
+
+    /**
+     * Set the folder name default value as the artifactId.
+     */
+    @Nullable
+    @Override
+    public ModuleWizardStep modifySettingsStep(@NotNull SettingsStep settingsStep) {
+        ModuleNameLocationSettings moduleNameLocationSettings = settingsStep.getModuleNameLocationSettings();
+        if (moduleNameLocationSettings != null && moduleDescriptor.getArtifactId() != null) {
+            moduleNameLocationSettings.setModuleName(StringUtil.sanitizeJavaIdentifier(moduleDescriptor.getArtifactId()));
+        }
+        return super.modifySettingsStep(settingsStep);
+    }
+
+    @Override
+    public void setupRootModel(@NotNull ModifiableRootModel rootModel) throws ConfigurationException {
+        super.setupRootModel(rootModel);
+        generateFromArchetype(rootModel.getProject(), rootModel.getSdk());
```

Comment:
Just remark:  `rootModel.getSdk()` can return null and will result in NPE later on. But I could not figure out in the IDE how to get there a null (setting the JDK to null in the IDE is not an issue and doesn't result in NPE).